File: 1894-merge-strings-alternately/solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        result = []

        for i in range(min(len(word1), len(word2))):
            result.append(word1[i])
            result.append(word2[i])
        
        if len(word1) < len(word2):
            result.append(word2[len(word1):])
        else:
            result.append(word1[len(word2):])

        return ''.join(result)

File: 1894-merge-strings-alternately/solution.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int len = Math.max(word1.length(), word2.length());
        int i = 0;
        StringBuilder sb = new StringBuilder();

        while (i < len) {
            if (i < word1.length()) {
                sb.append(word1.charAt(i));
            }

            if (i < word2.length()) {
                sb.append(word2.charAt(i));
            }

            i++;
        }

        return sb.toString();
    }
}

